feast_component/executor.py

```diff
@@ -16,13 +16,165 @@
+
+    if feature_list:
+        features = feature_list
+    elif feature_service:
+        features = fs.get_feature_service(feature_service)
+    else:
+        raise RuntimeError("Either feature service or feature list should be provided")
+
+    return fs.get_historical_features(entity_df=entity_query, features=features)
 
-  def GetInputSourceToExamplePTransform(self) -> beam.PTransform:
-    """Returns PTransform for Feast to TF examples."""
+
+def _get_gcp_project(exec_properties: Dict[str, Any]) -> Optional[str]:
+    # Get GCP project from exec_properties.
+    beam_pipeline_args = exec_properties["_beam_pipeline_args"]
+    pipeline_options = beam.options.pipeline_options.PipelineOptions(beam_pipeline_args)
+    # Try to parse the GCP project ID from the beam pipeline options.
+    project = pipeline_options.view_as(
+        beam.options.pipeline_options.GoogleCloudOptions
+    ).project
+    if isinstance(project, value_provider.ValueProvider):
+        return project.get()
     return None
+
+
+@beam.ptransform_fn
+@beam.typehints.with_input_types(beam.Pipeline)
+@beam.typehints.with_output_types(bytes)
+def _FeastToExampleTransform(
+    pipeline: beam.Pipeline, exec_properties: Dict[str, Any], split_pattern: str
+) -> beam.pvalue.PCollection:
+    """Read from BigQuery and transform to TF examples.
+
+    Args:
+      pipeline: beam pipeline.
+      exec_properties: A dict of execution properties.
+      split_pattern: Split.pattern in Input config, a BigQuery sql string.
+
+    Returns:
+      PCollection of TF examples.
+    """
+    # Load custom config dictionary
+    custom_config = _load_custom_config(exec_properties["custom_config"])
+
+    # Get Feast retrieval job
+    retrieval_job = _get_retrieval_job(
+        entity_query=split_pattern, custom_config=custom_config
+    )
+
+    # Setup datasource and converter.
+    if isinstance(retrieval_job, BigQueryRetrievalJob):
+        query = retrieval_job.to_sql()
+        datasource = utils.ReadFromBigQuery(query=query)
+        converter = converters._BQConverter(query, _get_gcp_project(exec_properties))
+    else:
+        raise NotImplementedError(
+            f"Support for {type(retrieval_job)} is not available yet. For now we only support BigQuery source."
+        )
+
+    # Setup converter from dictionary of str -> value to bytes
+    map_function = None
+    out_format = exec_properties.get("output_data_format", example_gen_pb2.FORMAT_TF_EXAMPLE)
+    if out_format == example_gen_pb2.FORMAT_TF_EXAMPLE:
+        map_function = converter.RowToExampleBytes
+    elif out_format == example_gen_pb2.FORMAT_TF_SEQUENCE_EXAMPLE:
+      map_function = converter.RowToSequenceExampleBytes
+    else:
+        raise NotImplementedError(
+            f"Format {out_format} is not currently supported. Currently we only support tfexample"
+        )
+
+    # Setup pipeline
+    return (
+        pipeline
+        | "DataSource" >> datasource
```

Comment:
`DataSource` -> `QueryTable`? Could not find a reference for `DataSource`, [reference for `QueryTable`](https://github.com/tensorflow/tfx/blob/master/tfx/extensions/google_cloud_big_query/example_gen/executor.py)